gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -490,33 +526,84 @@ var gpii = fluid.registerNamespace("gpii");
     };
 
     /**
+     * The lifecycleManager queue is used to hold the high-level actions that needs to happen,
+     * such as starting the login process, starting logout process, starting the update process.
+     * The entries in the queue are of the format { func: <functionToCall>, arg: <argument> } where
+     * <functionToCall> is a single-argument function that returns a promise. The promise should be resolved when
+     * the function is complete (including side-effects).
+     * The queue is run sequentially, and an item is considered "done" once the promise returned by its
+     * function is resolved.
+     */
+    gpii.lifecycleManager.addToQueue = function (that, queue, item) {
+        var newItem = fluid.copy(item);
+        newItem.promise = fluid.promise();
+        queue.push(newItem);
+        if (!that.isProcessingQueue) {
+            that.processQueue();
+        }
+        return newItem.promise;
+    };
+
+    gpii.lifecycleManager.clearQueue = function (queue) {
+        while (queue.length > 0) {
+            var item = queue.shift();
+            item.promise.reject("clearQueue: clearing lifecycleManager promise from queue - remaining items: " + queue.length);
+        }
+    };
+
+    gpii.lifecycleManager.processQueue = function (that, queue) {
+        // mark that we're currently processing queue
+        that.isProcessingQueue = true;
+        // pick first item and process
+        var item = queue.shift();
+        var func = fluid.makeInvoker(that, item.func, "");
```

Comment:
The idea was to send "args" into here as well, and then you get the work done for you of expanding and converting them